src/meltano/core/block/parser.py

```diff
@@ -290,11 +290,22 @@ def _find_next_elb_set(  # noqa: WPS231, WPS213
                     mapping=self._mappings_ref.get(next_block),
                     idx=next_block,
                 )
-                blocks.append(
-                    builder.make_block(
-                        plugin,
+                # Checks to see if the mapper plugin name is the same as the mappings name
+                # If they both match then a validation error is raised because the
+                # meltano run command needs the mappings name to obtain the settings to
+                # pass to the parent mapper plugin.  We also want to fail if the user names them
+                # the same to stop errors due to ambiguous commands.
+                if plugin.name == self._mappings_ref.get(next_block):
+                    self.log.warning("mapper plugin name used")
```

Comment:
```suggestion
                    self.log.warning("Found unexpected mapper plugin name." plugin_name=plugin_name)
```

Just adds a little more context (namely what plugin_name was seen)